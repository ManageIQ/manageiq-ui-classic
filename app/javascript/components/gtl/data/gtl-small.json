{
  "settings": {"perpage":5,"current":1,"items":407,"total":82,"sort_dir":"ASC","sort_col":0, "is_ascending":true},
  "data": {
    "head":[
      {"is_narrow":true},{"is_narrow":true},
      {"text":"Name","sort":"str","col_idx":0,"align":"left"},
      {"text":"Power State","sort":"str","col_idx":1,"align":"left"},
      {"text":"Provider","sort":"str","col_idx":2,"align":"left"},
      {"text":"Cluster","sort":"str","col_idx":3,"align":"left"},
      {"text":"Host","sort":"str","col_idx":4,"align":"left"},
      {"text":"IP Addresses","sort":"str","col_idx":5,"align":"left"},
      {"text":"Datastore","sort":"str","col_idx":6,"align":"left"},
      {"text":"Compliant","sort":"str","col_idx":7,"align":"left"},
      {"text":"Total Snapshots","sort":"str","col_idx":8,"align":"right"},
      {"text":"Allocated Size","sort":"str","col_idx":9,"align":"right"},
      {"text":"Last Analysis Time","sort":"str","col_idx":10,"align":"left"},
      {"text":"Region","sort":"str","col_idx":11,"align":"left"}
    ],
    "rows":[
      {
        "id":"1872","long_id":"1872",
        "cells":[
          {"is_checkbox":true},{"title":"View this item","image":"/assets/manageiq.svg"},{"text":"47d6e0c7-bd7b-4045-9a05-b875afddcc7c"},{"text":"orphaned"},{"text":""},{"text":""},{"text":""},{"text":""},{"text":"Shared-NFS"},{"text":""},{"text":"0"},{"text":"40 GB"},{"text":""},{"text":"Region 0"}
        ],
        "quad":{
          "top_left":{"tooltip":"linux redhat","fileicon":"/assets/manageiq.svg"},
          "top_right":{"tooltip":"orphaned","fonticon":"ff ff-orphaned","background":"#336699"},
          "bottom_left":{"fileicon":"/assets/manageiq.svg","tooltip":"Virtual Machine (VMware)"},
          "bottom_right":{"text":"0","tooltip":"0 Snapshots"},
          "middle":null
        },
        "tree_id":"v-1872"
      },
      {"id":"2199","long_id":"2199","cells":[{"is_checkbox":true},{"title":"View this item","image":"/assets/manageiq.svg"},{"text":"aaaop3322"},{"text":"archived"},{"text":""},{"text":""},{"text":""},{"text":""},{"text":""},{"text":""},{"text":"0"},{"text":"16 GB"},{"text":""},{"text":"Region 0"}],"quad":{"top_left":{"tooltip":"linux redhat","fileicon":"/assets/manageiq.svg"},"top_right":{"tooltip":"archived","fonticon":"fa fa-archive","background":"#336699"},"bottom_left":{"fileicon":"/assets/manageiq.svg","tooltip":"Virtual Machine (VMware)"},"bottom_right":{"text":"0","tooltip":"0 Snapshots"},"middle":null},"tree_id":"v-2199"},
      {"id":"2080","long_id":"2080","cells":[{"is_checkbox":true},{"title":"View this item","image":"/assets/manageiq.svg"},{"text":"aaaop333"},{"text":"archived"},{"text":""},{"text":""},{"text":""},{"text":""},{"text":""},{"text":""},{"text":"0"},{"text":"16 GB"},{"text":"12/21/17 19:53:45 UTC"},{"text":"Region 0"}],"quad":{"top_left":{"tooltip":"linux redhat","fileicon":"/assets/manageiq.svg"},"top_right":{"tooltip":"archived","fonticon":"fa fa-archive","background":"#336699"},"bottom_left":{"fileicon":"/assets/manageiq.svg","tooltip":"Virtual Machine (VMware)"},"bottom_right":{"text":"0","tooltip":"0 Snapshots"},"middle":null},"tree_id":"v-2080"},
      {"id":"2079","long_id":"2079","cells":[{"is_checkbox":true},{"title":"View this item","image":"/assets/manageiq.svg"},{"text":"aaop001"},{"text":"archived"},{"text":""},{"text":""},{"text":""},{"text":""},{"text":""},{"text":""},{"text":"0"},{"text":"16 GB"},{"text":""},{"text":"Region 0"}],"quad":{"top_left":{"tooltip":"linux redhat","fileicon":"/assets/manageiq.svg"},"top_right":{"tooltip":"archived","fonticon":"fa fa-archive","background":"#336699"},"bottom_left":{"fileicon":"/assets/manageiq.svg","tooltip":"Virtual Machine (VMware)"},"bottom_right":{"text":"0","tooltip":"0 Snapshots"},"middle":null},"tree_id":"v-2079"},
      {"id":"66","long_id":"66","cells":[{"is_checkbox":true},{"title":"View this item","image":"/assets/manageiq.svg"},{"text":"amaya1"},{"text":"archived"},{"text":""},{"text":""},{"text":""},{"text":""},{"text":""},{"text":""},{"text":"0"},{"text":"16 GB"},{"text":""},{"text":"Region 0"}],"quad":{"top_left":{"tooltip":"linux redhat","fileicon":"/assets/manageiq.svg"},"top_right":{"tooltip":"archived","fonticon":"fa fa-archive","background":"#336699"},"bottom_left":{"fileicon":"/assets/manageiq.svg","tooltip":"Virtual Machine (VMware)"},"bottom_right":{"text":"0","tooltip":"0 Snapshots"},"middle":null},"tree_id":"v-66"}
    ]
  },
  "messages":null
}
