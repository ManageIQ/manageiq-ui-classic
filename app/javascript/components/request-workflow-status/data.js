/** Types of workflow state status */
export const workflowStateTypes = {
  success: { text: 'success', tagType: 'green' },
  error: { text: 'error', tagType: 'red' },
  failure: { text: 'failure', tagType: 'red' },
  failed: { text: 'failed', tagType: 'gray' },
  pending: { text: 'pending', tagType: 'gray' },
  running: { text: 'running', tagType: 'gray' },
};

/** Function to get the header data of workflow states table. */
const headerData = () => ([
  {
    key: 'name',
    header: __('Name'),
  },
  {
    key: 'enteredTime',
    header: __('Entered Time'),
  },
  {
    key: 'finishedTime',
    header: __('Finished Time'),
  },
  {
    key: 'duration',
    header: __('Duration'),
  },
]);

// const convertDate = (date) => `${moment(date).format('MM/DD/YYYY')} ${moment(date).format('h:mm:ss A')}`;
const convertDate = (date) => {
  const utcDate = new Date(date);
  const year = utcDate.getUTCFullYear();
  const month = (utcDate.getUTCMonth() + 1).toString().padStart(2, '0');
  const day = utcDate.getUTCDate().toString().padStart(2, '0');
  const hours = utcDate.getUTCHours();
  const minutes = utcDate.getUTCMinutes().toString().padStart(2, '0');
  const period = hours < 12 ? 'AM' : 'PM';
  const hours12 = hours % 12 || 12; // Convert 0 to 12 for 12:00 AM

  const formattedDate = `${month}/${day}/${year} ${hours12}:${minutes} ${period}`;
  return formattedDate.toString();
};

// Converts the duration time in ms and returns a string in format: w days x hours y minutes z seconds
// duration: time in ms
const convertDuration = (duration) => {
  const durationString = moment.duration(duration, 'milliseconds').toISOString().split('PT')[1];
  let startIndex = 0;
  let resultString = '';
  if (durationString.indexOf('H') >= 0) {
    resultString += `${durationString.slice(startIndex, durationString.indexOf('H'))}h `;
    startIndex = durationString.indexOf('H') + 1;
  }
  if (durationString.indexOf('M') >= 0) {
    resultString += `${durationString.slice(startIndex, durationString.indexOf('M'))}m `;
    startIndex = durationString.indexOf('M') + 1;
  }
  if (durationString.indexOf('S') >= 0) {
    resultString += `${durationString.slice(startIndex, durationString.indexOf('S'))}s`;
    startIndex = durationString.indexOf('S') + 1;
  }
  return resultString;
};

const getItemIcon = (item) => {
  if (item.FinishedTime) {
    if (item.Output && item.Output.Error) {
      if (item.RetryCount) {
        return { icon: 'carbon--RetryFailed' }
      } else {
        return { icon: 'carbon--MisuseOutline' }
      }
    } else {
      return { icon: 'carbon--CheckmarkOutline' }
    }
  } else {
    return { icon: 'carbon--PlayOutline' }
  }
};

/** Function to get the row data of workflow states table. */
const rowData = ({ StateHistory }) => StateHistory.map((item) => ({
  id: item.Guid.toString(),
  name: { text: item.Name, ...getItemIcon(item) },
  enteredTime: convertDate(item.EnteredTime.toString()),
  finishedTime: convertDate(item.FinishedTime.toString()),
  duration: convertDuration(item.Duration * 1000),
}));

/** Function to return the header, row and status data required for the RequestWorkflowStatus component.  */
export const workflowStatusData = (response) => {
  const type = 'ManageIQ::Providers::Workflows::AutomationManager::WorkflowInstance';
  if (response.type !== type) {
    return undefined;
  }
  const rows = response.context ? rowData(response.context) : [];
  if (response.context && response.context.State && !response.context.State.FinishedTime) {
    const state = response.context.State;
    const currentTime = new Date(); // Date Object for current time
    const oldTime = Date.parse(state.EnteredTime); // ms since start time to entered time in UTC
    const durationTime = currentTime.getTime() - oldTime; // ms from entered time to current time

    rows.push({
      id: state.Guid.toString(),
      name: { text: state.Name, ...getItemIcon(state)  },
      enteredTime: convertDate(state.EnteredTime.toString()),
      finishedTime: '',
      duration: convertDuration(durationTime),
    });
  }
  const headers = headerData();
  const name = response.name || response.description;
  return {
    headers, rows, status: response.status, name, parentId: response.parent_id, id: response.id, type: response.type,
  };
};
