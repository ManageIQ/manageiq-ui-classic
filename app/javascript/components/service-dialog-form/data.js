import React from 'react';
import {
  CheckboxChecked32, RadioButtonChecked32, Time32, StringText32, TextSmallCaps32, CaretDown32, Tag32, Calendar32,
} from '@carbon/icons-react';

export const dragItems = {
  COMPONENT: 'component',
  SECTION: 'section',
  FIELD: 'field',
};

/** Data needed to render the dynamic components on the left hand side of the form. */
export const dynamicComponents = [
  { id: 1, title: 'Text Box', icon: <StringText32 /> },
  { id: 2, title: 'Text Area', icon: <TextSmallCaps32 /> },
  { id: 3, title: 'Check Box', icon: <CheckboxChecked32 /> },
  { id: 4, title: 'Dropdown', icon: <CaretDown32 /> },
  { id: 5, title: 'Radio Button', icon: <RadioButtonChecked32 /> },
  { id: 6, title: 'Datepicker', icon: <Calendar32 /> },
  { id: 7, title: 'Timepicker', icon: <Time32 /> },
  { id: 8, title: 'Tag Control', icon: <Tag32 /> },
];

/** Function which returens the default data for a section under a tab. */
export const defaultSectionContents = (tabId, sectionId) => ({
  tabId,
  sectionId,
  title: 'New Section',
  fields: [],
  order: 0,
});

/** Function which returns the default data for a tab with default section. */
export const defaultTabContents = (tabId) => ({
  tabId,
  name: tabId === 0 ? __('New Tab') : __(`New Tab ${tabId}`),
  sections: [defaultSectionContents(tabId, 0)],
});

/** Function to create a dummy tab for creating new tabs. */
export const createNewTab = () => ({
  tabId: 'new',
  name: 'Create Tab',
  sections: [],
});
