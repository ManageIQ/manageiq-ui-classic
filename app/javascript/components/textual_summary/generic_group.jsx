import * as React from 'react';
import PropTypes from 'prop-types';
import MiqStructuredList from '../miq-structured-list';

export default function GenericGroup(props) {
  const { title, items } = props;
  return (
    <MiqStructuredList
      rows={items}
      title={title}
      mode="generic_group"
      onClick={(item, event) => props.onClick(item, event)}
    />
  );
}

GenericGroup.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.any).isRequired,
  onClick: PropTypes.func.isRequired,
};
