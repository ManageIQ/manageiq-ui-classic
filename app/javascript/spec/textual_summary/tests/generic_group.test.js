import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GenericGroup from '../../../components/textual_summary/generic_group';
import { genericGroupData } from '../data/generic_group';

describe('GenericGroup', () => {
  it('renders title and rows', () => {
    const wrapper = shallow(<GenericGroup
      items={genericGroupData.items}
      title={genericGroupData.title}
      onClick={() => null}
    />);
    expect(wrapper.html()).toContain(genericGroupData.title);
  });

  it('renders just fine', () => {
    const group = mount(<GenericGroup
      items={genericGroupData.items}
      title={genericGroupData.title}
      onClick={() => null}
    />);
    expect(toJson(group)).toMatchSnapshot();
  });
});
