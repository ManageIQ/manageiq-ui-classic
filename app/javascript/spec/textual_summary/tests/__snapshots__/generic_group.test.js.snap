// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericGroup renders just fine 1`] = `
<GenericGroup
  items={
    Array [
      Object {
        "hoverClass": "bar",
        "label": "foo",
        "value": "bar",
      },
      Object {
        "hoverClass": "bar",
        "label": "mrkef?",
        "value": "v zime!",
      },
    ]
  }
  onClick={[Function]}
  title="Foo Bar"
>
  <MiqStructuredList
    headers={Array []}
    mode="generic_group"
    onClick={[Function]}
    rows={
      Array [
        Object {
          "hoverClass": "bar",
          "label": "foo",
          "value": "bar",
        },
        Object {
          "hoverClass": "bar",
          "label": "mrkef?",
          "value": "v zime!",
        },
      ]
    }
    title="Foo Bar"
  >
    <Accordion
      align="start"
      className="miq-structured-list-accordion generic_group"
    >
      <ul
        className="bx--accordion miq-structured-list-accordion generic_group bx--accordion--start bx--accordion--md"
      >
        <AccordionItem
          open={true}
          title="Foo Bar"
        >
          <li
            className="bx--accordion__item bx--accordion__item--active"
            onAnimationEnd={[Function]}
          >
            <defaultRenderToggle
              aria-controls="accordion-item-1"
              aria-expanded={true}
              className="bx--accordion__heading"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <button
                aria-controls="accordion-item-1"
                aria-expanded={true}
                className="bx--accordion__heading"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <ForwardRef(ChevronRight16)
                  className="bx--accordion__arrow"
                >
                  <Icon
                    className="bx--accordion__arrow"
                    fill="currentColor"
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <svg
                      aria-hidden={true}
                      className="bx--accordion__arrow"
                      fill="currentColor"
                      focusable="false"
                      height={16}
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                      />
                    </svg>
                  </Icon>
                </ForwardRef(ChevronRight16)>
                <Text
                  as="div"
                  className="bx--accordion__title"
                >
                  <div
                    className="bx--accordion__title"
                    dir="auto"
                  >
                    Foo Bar
                  </div>
                </Text>
              </button>
            </defaultRenderToggle>
            <div
              className="bx--accordion__content"
              id="accordion-item-1"
            >
              <FeatureToggle(StructuredListWrapper)
                ariaLabel="Structured list"
                className="miq-structured-list generic_group"
              >
                <StructuredListWrapper
                  ariaLabel="Structured list"
                  className="miq-structured-list generic_group"
                  isCondensed={false}
                  isFlush={false}
                  selection={false}
                >
                  <div
                    aria-label="Structured list"
                    className="bx--structured-list miq-structured-list generic_group"
                    role="table"
                  >
                    <FeatureToggle(StructuredListBody)>
                      <StructuredListBody
                        onKeyDown={[Function]}
                      >
                        <div
                          className="bx--structured-list-tbody"
                          onKeyDown={[Function]}
                          role="rowgroup"
                        >
                          <FeatureToggle(StructuredListRow)
                            className=""
                            key="0"
                            onClick={[Function]}
                            tabIndex={0}
                            title=""
                          >
                            <StructuredListRow
                              className=""
                              head={false}
                              label={false}
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              tabIndex={0}
                              title=""
                            >
                              <div
                                className="bx--structured-list-row"
                                onClick={[Function]}
                                role="row"
                                title=""
                              >
                                <FeatureToggle(StructuredListCell)
                                  className="label_header"
                                >
                                  <StructuredListCell
                                    className="label_header"
                                    head={false}
                                    noWrap={false}
                                  >
                                    <div
                                      className="label_header bx--structured-list-td"
                                      role="cell"
                                    >
                                      foo
                                    </div>
                                  </StructuredListCell>
                                </FeatureToggle(StructuredListCell)>
                                <FeatureToggle(StructuredListCell)
                                  className="content_value object_item"
                                  title=""
                                >
                                  <StructuredListCell
                                    className="content_value object_item"
                                    head={false}
                                    noWrap={false}
                                    title=""
                                  >
                                    <div
                                      className="content_value object_item bx--structured-list-td"
                                      role="cell"
                                      title=""
                                    >
                                      <div
                                        className="content"
                                      >
                                        <div
                                          className="expand wrap_text"
                                        >
                                          bar
                                        </div>
                                      </div>
                                    </div>
                                  </StructuredListCell>
                                </FeatureToggle(StructuredListCell)>
                              </div>
                            </StructuredListRow>
                          </FeatureToggle(StructuredListRow)>
                          <FeatureToggle(StructuredListRow)
                            className=""
                            key="1"
                            onClick={[Function]}
                            tabIndex={1}
                            title=""
                          >
                            <StructuredListRow
                              className=""
                              head={false}
                              label={false}
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              tabIndex={1}
                              title=""
                            >
                              <div
                                className="bx--structured-list-row"
                                onClick={[Function]}
                                role="row"
                                title=""
                              >
                                <FeatureToggle(StructuredListCell)
                                  className="label_header"
                                >
                                  <StructuredListCell
                                    className="label_header"
                                    head={false}
                                    noWrap={false}
                                  >
                                    <div
                                      className="label_header bx--structured-list-td"
                                      role="cell"
                                    >
                                      mrkef?
                                    </div>
                                  </StructuredListCell>
                                </FeatureToggle(StructuredListCell)>
                                <FeatureToggle(StructuredListCell)
                                  className="content_value object_item"
                                  title=""
                                >
                                  <StructuredListCell
                                    className="content_value object_item"
                                    head={false}
                                    noWrap={false}
                                    title=""
                                  >
                                    <div
                                      className="content_value object_item bx--structured-list-td"
                                      role="cell"
                                      title=""
                                    >
                                      <div
                                        className="content"
                                      >
                                        <div
                                          className="expand wrap_text"
                                        >
                                          v zime!
                                        </div>
                                      </div>
                                    </div>
                                  </StructuredListCell>
                                </FeatureToggle(StructuredListCell)>
                              </div>
                            </StructuredListRow>
                          </FeatureToggle(StructuredListRow)>
                        </div>
                      </StructuredListBody>
                    </FeatureToggle(StructuredListBody)>
                  </div>
                </StructuredListWrapper>
              </FeatureToggle(StructuredListWrapper)>
            </div>
          </li>
        </AccordionItem>
      </ul>
    </Accordion>
  </MiqStructuredList>
</GenericGroup>
`;
